library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c

```diff
@@ -49,8 +49,8 @@ libspdm_return_t libspdm_encode_secured_message(
     size_t record_header_size;
     spdm_secured_message_cipher_header_t *enc_msg_header;
     bool result;
-    uint8_t key[LIBSPDM_MAX_AEAD_KEY_SIZE];
-    uint8_t salt[LIBSPDM_MAX_AEAD_IV_SIZE];
+    const uint8_t *key;
+    uint8_t *salt;
```

Comment:
The salt should stay `const`, like the key.
Also we'll need to declare `uint8_t iv[LIBSPDM_MAX_AEAD_IV_SIZE]`, that actually gets fed into `libspdm_aead_encryption`.

Reply:
I don't think we need define `uint8_t iv[LIBSPDM_MAX_AEAD_IV_SIZE]`. That will go back to original problem.